tests/test_api.py

```diff
@@ -427,6 +427,68 @@ def test_exception_raised_upon_catalogue_import_with_same_uuid_but_different_eve
         with self.assertRaises(ValueError):
             import_json(export_blob)
 
+    def test_export_import_multiple_catalogues_with_shared_and_individual_events(self):
+        shared_events = [generate_event() for _ in range(2)]
+
+        events1 = [generate_event() for _ in range(2)]
+        events2 = [generate_event() for _ in range(2)]
+
+        catalogue1 = create_catalogue("TestExportImportCatalogue1", "Patrick",
+                                      events=events1 + shared_events)
+        catalogue2 = create_catalogue("TestExportImportCatalogue2", "Patrick",
+                                      events=events2 + shared_events)
+
+        export_blob = export_json([catalogue1, catalogue2])
+
+        discard()
+
+        import_json(export_blob)
+
+        def s(l):
+            return sorted(l, key=lambda x: x.uuid)
+
+        assert s(events1 + shared_events + events2) == s(get_events())
+        assert s(events1 + shared_events) == s(get_events(catalogue1))
+        assert s(events2 + shared_events) == s(get_events(catalogue2))
+
+        assert s([catalogue1, catalogue2]) == s(get_catalogues())
+
+    def test_import_multiple_catalogues_with_shared_and_individual_but_already_existing_events(self):
+        shared_events = [generate_event() for _ in range(2)]
+
+        events1 = [generate_event() for _ in range(2)]
+        events2 = [generate_event() for _ in range(2)]
+
+        catalogue1 = create_catalogue("TestExportImportCatalogue1", "Patrick",
+                                      events=events1 + shared_events)
+        catalogue2 = create_catalogue("TestExportImportCatalogue2", "Patrick",
+                                      events=events2 + shared_events)
+
+        export_blob = export_json([catalogue1, catalogue2])
+
+        import_json(export_blob)
+
+        def s(l):
+            return sorted(l, key=lambda x: x.uuid)
+
+        assert s(events1 + shared_events + events2) == s(get_events())
+        assert s(events1 + shared_events) == s(get_events(catalogue1))
+        assert s(events2 + shared_events) == s(get_events(catalogue2))
+
+        assert s([catalogue1, catalogue2]) == s(get_catalogues())
+
+    def test_export_import_of_existing_dynamic_catalogue(self):
+        events = [generate_event() for _ in range(2)]
+        events[0].author = "Patrick"
+        events[1].author = "Alexis"
+
+        c = create_catalogue("Events_of_Alexis", "Patrick", predicate=Comparison('==', Field('author'), 'Alexis'))
+
+        assert get_events(c) == [events[1]]
+
+        export_blob = export_json(c)
+        import_json(export_blob)
+
```

Comment:
That's weird how the test finishes without any assert while an export/import cycle was performed.

Reply:
Yes, the test fixes a bug, before the fix, the call to `import_json()` threw an exception. It should not. Now it does not anymore.

Reply:
Ok, I understand now, then this could be renamed to  `test_of_existing_dynamic_catalogue_doesnt_crash` and also I see some additional checks here such as checking db state -> same number of catalogs and events.

Reply:
Done.